rcl_action/test/rcl_action/test_action_communication.cpp

```diff
@@ -1106,3 +1106,61 @@ TEST_F(CLASSNAME(TestActionCommunication, RMW_IMPLEMENTATION), test_invalid_stat
   ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
   action_msgs__msg__GoalStatusArray__fini(&incoming_status_array);
 }
+
+TEST_F(CLASSNAME(TestActionCommunication, RMW_IMPLEMENTATION), test_valid_feedback_comm_maybe_fail)
+{
+  test_msgs__action__Fibonacci_FeedbackMessage outgoing_feedback;
+  test_msgs__action__Fibonacci_FeedbackMessage incoming_feedback;
+  test_msgs__action__Fibonacci_FeedbackMessage__init(&outgoing_feedback);
+  test_msgs__action__Fibonacci_FeedbackMessage__init(&incoming_feedback);
+
+  // Initialize feedback
+  ASSERT_TRUE(
+    rosidl_runtime_c__int32__Sequence__init(
+      &outgoing_feedback.feedback.sequence, 3));
+  outgoing_feedback.feedback.sequence.data[0] = 0;
+  outgoing_feedback.feedback.sequence.data[1] = 1;
+  outgoing_feedback.feedback.sequence.data[2] = 2;
+  init_test_uuid0(outgoing_feedback.goal_id.uuid);
+
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    // Publish feedback with valid arguments
+    rcl_ret_t ret = rcl_action_publish_feedback(&this->action_server, &outgoing_feedback);
+    if (RCL_RET_OK != ret) {
+      continue;
```

Comment:
Same about `rcl_reset_error()` here and below

Reply:
Added.